File: frontend/src/api/rag.ts
import client from "./client";

export interface QueryResponse {
  answer: string;
  sources: { content: string }[];
}

export const ragApi = {
  query: (kbId: number, question: string, topK = 5) =>
    client.post<QueryResponse>(`/knowledge-bases/${kbId}/query`, {
      question,
      top_k: topK,
    }),

  /** 多知识库查询，kb_ids 为空表示全部知识库 */
  batchQuery: (question: string, kbIds: number[] = [], topK = 5) =>
    client.post<QueryResponse>("/knowledge-bases/query", {
      question,
      top_k: topK,
      kb_ids: kbIds,
    }),
};

File: backend/app/schemas/rag.py
"""RAG 相关模式"""
from pydantic import BaseModel, Field


class QueryRequest(BaseModel):
    question: str = Field(min_length=1, max_length=4000)
    top_k: int = Field(default=5, ge=1, le=20)


class BatchQueryRequest(BaseModel):
    question: str = Field(min_length=1, max_length=4000)
    top_k: int = Field(default=5, ge=1, le=20)
    kb_ids: list[int] = Field(default_factory=list)  # 空表示全部知识库


class QueryResponse(BaseModel):
    answer: str
    sources: list[dict] = Field(default_factory=list)
